aptible/data_source_environment_test.go

```diff
@@ -0,0 +1,70 @@
+package aptible
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccDataSourceEnvironment_validation(t *testing.T) {
+	requiredAttrs := []string{"handle"}
+	var testSteps []resource.TestStep
+
+	for _, attr := range requiredAttrs {
+		testSteps = append(testSteps, resource.TestStep{
+			PlanOnly:    true,
+			Config:      `data "aptible_environment" "test" {}`,
+			ExpectError: regexp.MustCompile(fmt.Sprintf("%q is required", attr)),
+		})
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviderFactories,
+		Steps:             testSteps,
+	})
+}
+
+func TestAccDataSourceEnvironment_deploy(t *testing.T) {
+	rHandle := acctest.RandString(10)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviderFactories,
+		CheckDestroy:      testAccCheckEnvironmentDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAptibleEnvironment(rHandle),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("aptible_environment.test", "handle", rHandle),
+					resource.TestCheckResourceAttr("aptible_environment.test", "org_id", testOrganizationId),
+					resource.TestCheckResourceAttr("aptible_environment.test", "stack_id", strconv.Itoa(testStackId)),
+				),
+			}, {
+				ResourceName:            "aptible_environment.test",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"org_id", "stack_id"},
+			},
+			{
+				Config: testDataAccAptibleEnvironment(rHandle),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.aptible_environment.test", "handle", rHandle),
+				),
```

Comment:
Since `env_id` is the attribute we care about, it would be good to check for it as well. Unfortunately, I don't think there's an easy way to get values off of resources from other steps but if you use the `APTIBLE_ENVIRONMENT_ID` to test the data source without creating an `aptible_environment` resource then you can retrieve the object from the API and test that the data source is returning the same values.

Reply:
It seems that you do something similar for the `aptible_stack` data source.

Reply:
In light of the issue with TF_ACC not actually preventing the test initializaiton from running I'm not worried about this right now.